lib/versioning/pep440/range.spec.ts

```diff
@@ -0,0 +1,14 @@
+import { checkRangeAndRemoveUnnecessaryRangeLimit } from './range';
+
+test.each`
+  rangeInput           | expected
+  ${'==4.1.*,>=3.2.2'} | ${'==4.1.*'}
+  ${'==4.0.*,>=3.2.2'} | ${'==4.0.*'}
+  ${'~=1.2.3,!=1.1.1'} | ${'~=1.2.3,!=1.1.1'}
```

Comment:
```suggestion
  ${'~=1.2.3,!=1.1.1'} | ${'~=1.2.3'}
```

Can this be done?

Reply:
yes, sure. 
but I think this change will effect the previous tests(I need to change the expected result for each case).
<img width="381" alt="image" src="https://user-images.githubusercontent.com/98313426/153744197-1d71e433-35fc-4b32-8077-4093af1b53bc.png">

Reply:
Feel free to share what you think the new results will be in advance, before you write the code

Reply:
hi,
Regarding the changes you suggested, when we have rangeInput= ${'~ =1.2.3,!=1.1.1'} the result should be  ${'~ =1.2.3'}, 
as a result of this change I need to make some changes in previous unit tests expected results for example the tests I shared above. 
According to the documentation:
`>=1.2.3, ==1.2.* <=> ~=1.2.3`
in case the rangeInput  = ${'~=1.2.3,!=1.1.1'} expected result this alone ~=1.2.3 since 1.1.1 is not even in the range.
So the versioning behavior will be changed and the test cases will be changed too in accordance.

This is not related to my changes I will open another PR for refactoring this issue on its own.